### PR TITLE
Cleans the backend logic for headshots

### PR DESCRIPTION
**Pull request by ZephyrTFA**
Also cleans up the underlying logic

### CODE REVIEW

**Comment by Zonespace27** [comment]
Imgur commonly didn't work when I was creating the PR originally, are you sure it works?

(Also thanks for cleaning up my oldcode)

File: modular_skyrat/master_files/code/modules/client/preferences/headshot.dm [hunk 1]
```diff
@@ -19,18 +15,29 @@
...
+	if(!(extension in valid_extensions))
+		to_chat(usr, span_warning("The image must be one of the following extensions: '[english_list(valid_extensions)]'"))
 		return
-	if(!stored_link[usr?.ckey])
-		stored_link[usr?.ckey] = null
-	if(stored_link[usr?.ckey] != value)
+
+	find_index = findtext(value, link_regex)
+	if(find_index != 9)
+		to_chat(usr, span_warning("The image must be hosted on one of the following sites: 'Gyazo, Discord, Imjur'"))
```
Comment (tf-4) [inline modular_skyrat/master_files/code/modules/client/preferences/headshot.dm]:
```suggestion
		to_chat(usr, span_warning("The image must be hosted on one of the following sites: 'Gyazo, Discord, Imgur'"))
```

**Review by Useroth — APPROVED** [review]

**Comment by GoldenAlpharex** [comment]
Imjur

**Review by Zonespace27 — APPROVED** [review]
Tested locally

**Pull request merged**